src/screen/show_refs.rs

```diff
@@ -18,57 +22,122 @@ pub(crate) fn create(config: Rc<Config>, repo: Rc<Repository>, size: Rect) -> Re
-                        );
-
-                        let prefix = Span::raw(
-                            if branch.get().name() == head.as_ref().and_then(|h| h.name()) {
-                                "* "
-                            } else {
-                                "  "
-                            },
-                        );
-
-                        let upstream_name = if let Ok(upstream) = branch.upstream() {
-                            if let Ok(Some(name)) = upstream.name() {
-                                Span::styled(name.to_string(), &style.remote)
-                            } else {
-                                Span::raw("")
-                            }
-                        } else {
-                            Span::raw("")
-                        };
-
-                        Item {
-                            id: name.clone().content,
-                            display: Line::from(vec![
-                                prefix,
-                                name.clone(),
-                                Span::raw("   "),
-                                upstream_name,
-                            ]),
-                            depth: 1,
-                            target_data: Some(TargetData::Branch(name.content.into())),
-                            ..Default::default()
-                        }
-                    }),
-            )
+            .chain(create_references_section(
+                &repo,
+                Reference::is_branch,
+                &style.branch,
+            )?)
+            .chain(create_remotes_sections(
+                &repo,
+                &style.section_header,
+                &style.remote,
+            )?)
+            .chain(iter::once(Item {
+                id: "tags".into(),
+                display: Line::styled("Tags".to_string(), &style.section_header),
+                section: true,
+                depth: 0,
+                ..Default::default()
+            }))
+            .chain(create_references_section(
+                &repo,
+                Reference::is_tag,
+                &style.tag,
+            )?)
             .collect())
         }),
     )
 }
+
+fn create_remotes_sections<'a>(
+    repo: &'a Repository,
+    header_style: &'a StyleConfigEntry,
+    item_style: &'a StyleConfigEntry,
+) -> Res<impl Iterator<Item = Item> + 'a> {
+    let all_remotes = create_references_section(repo, Reference::is_remote, item_style)?;
+    let mut remotes = HashMap::new();
+    for remote in all_remotes {
+        let name = match remote.id.split_once('/') {
+            None => remote.id.as_ref(),
+            Some((name, _)) => name,
+        };
+
+        match remotes.entry(name.to_string()) {
+            Entry::Vacant(entry) => {
+                entry.insert(vec![remote]);
+            }
+            Entry::Occupied(mut entry) => {
+                entry.get_mut().push(remote);
+            }
+        }
+    }
```

Comment:
There's an issue (since it's a hashmap). The order of the items will be randomized on a refresh.

These functions exist in git2, might be a nicer way than splitting by `/`:
```rust
repo.remotes()?.iter().map(|remote_name: Option<&str>| {
    // Why does find_remote accept a &str, when remote_name might be None?
    let remote = repo.find_remote(remote_name)?;
    todo!();
});

repo.remotes()?.iter_bytes().map(|remote_name: &[u8]| {
    // The bytes variant is non-optional, but how to get the data on a remote from here?
});
```

The git2 API is a bit messy sometimes. I'd like to start experimenting with Gitoxide some time soon.

Reply:
Done